### PR TITLE
SSCS-7574 Pact tests Configuration

### PR DESCRIPTION
**Pull request by danhamilton**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@Library("Infrastructure")
+
+def type = "java"          // supports "java", "nodejs" and "angular"
+
+def product = "idam-java-client"
+
+def component = "idam-java-client" // must match infrastructure module name
+
+withPipeline(type, product, component) {
```
Comment (timja) [inline Jenkinsfile_CNP:11]: you probably don't want to use withPipeline it's not designed for this,

you'll need to copy the required code in to the jenkinsfile

**Comment by timja** [comment]
The pact is failing when verified by Idam currently:

image

https://pact-broker.platform.hmcts.net/

File: src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        params.put("forename","emCaseOfficer");
+        params.put("surname", "jar123");
+        params.put("roles", rolesArray);
+
+        return builder
+                .given("a user exists", params)
+                .uponReceiving("Provider takes user/pwd and returns Auth code to Idam Client")
+                .path(IDAM_OPENID_TOKEN_URL)
+                .method(HttpMethod.POST.toString())
+                .body("redirect_uri=https%3A%2F%2Flocalhost%3A5000%2Freceiver&client_id=bsp&grant_type=password"
```
Comment (timja) [inline src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java:64]: will this work when idam tries to verify the pact?

Comment (danhamilton) [reply src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java]: Now working when running Pact provider verify for idam-api.

image

Comment (timja) [reply src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java]: Failed 23 hours ago but maybe need idam to re-run verification?

image

Comment (danhamilton) [reply src/contractTest/java/uk.gov.hmcts.reform.idam.client/IdamClientConsumerTest.java]: So the above was running provider pact verification locally against idam-api. The test I've added is on par with the em-annotations one which is passing. Based on that I'm confident the test as of this today is correct.

Making publishing a consumer pact test PR dependent dependent on provider running and verifying against prod pipeline definitely feels like a bad pattern. If someone wants to extend a provider verification step for consumer prs that's fine but beyond scope of this PR

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!groovy
+
+@Library("Infrastructure@SSCS-7575")
```
Comment (timja) [inline Jenkinsfile_CNP:3]: don't forget to update this when the jenkins-library pr is merged

**Pull request merged**